src/display-output.cc

```diff
@@ -27,36 +27,44 @@
 #include <config.h>
 
 #include "display-output.hh"
-#include "logging.h"
 
 #include <algorithm>
 #include <iostream>
 #include <sstream>
 #include <unordered_map>
 
 namespace conky {
-namespace {
 
-typedef std::unordered_map<std::string, display_output_base *>
-    display_outputs_t;
-
-/*
- * We cannot construct this object statically, because order of object
- * construction in different modules is not defined, so register_source could be
- * called before this object is constructed. Therefore, we create it on the
- * first call to register_source.
- */
-display_outputs_t *display_outputs;
-
-}  // namespace
-
-// HACK: force the linker to link all the objects in with test enabled
-extern void init_console_output();
-extern void init_ncurses_output();
-extern void init_file_output();
-extern void init_http_output();
-extern void init_x11_output();
-extern void init_wayland_output();
```

Comment:
And HACK gets replaced by [explicit template specialization](https://en.cppreference.com/w/cpp/language/template_specialization), which is a well supported language mechanism.